savefile/savefile.go

```diff
@@ -0,0 +1,129 @@
+package savefile
+
+import (
+	"bytes"
+	"compress/zlib"
+	"encoding/binary"
+	"errors"
+	"io"
+	"io/ioutil"
+
+	"github.com/sector-f/jhmod/nvc"
+)
+
+const (
+	// Magic header for Jupiter Hell save save files.
+	magic = "\xde\xc0\xad\xde"
+)
+
+type savedata struct {
+	// Player name
+	PlayerName string
+	// Game mode.  This can be "jh", and various others.
+	GameMode string
+	// The current level's name.
+	CurrentLevel string
+	// The seed used to generate the game.
+	Seed uint32
+}
+
+// Something was reading for data but got an EOF.
+var UnexpectedEndOfFileErr error = errors.New("unexpected eof")
+
+// General parse error.  TODO make more descriptive errors.
+var SaveParseErr error = errors.New("general parse error")
+
+// Read in a string from a buffer of size suffixed with a length byte.
+func readString(r io.Reader, size int) (string, error) {
+	buf := make([]byte, size)
+	nRead, err := r.Read(buf[:])
+	if err != nil {
+		return "", err
+	}
+	if nRead != size {
+		return "", UnexpectedEndOfFileErr
+	}
+	recordedSize := buf[size-1]
```

Comment:
Nit: I think the intended meaning ("The last item in the slice") would be more obvious as `buf[len(buf)-1]`

Reply:
made another func to help document it